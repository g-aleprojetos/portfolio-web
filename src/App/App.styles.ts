import styled, {keyframes} from 'styled-components';

const spin = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

export const App = styled.div`
  text-align: center;
`;

export const Header = styled.header`
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
`;

export const Link = styled.a`
  color: #61dafb;
`;

export const Logo = styled.img`
  height: 40vmin;
  pointer-events: none;

  @media (prefers-reduced-motion: no-preference) {
    animation: ${spin} infinite 20s linear;
  }
`;

export const Texto = styled.p``;
