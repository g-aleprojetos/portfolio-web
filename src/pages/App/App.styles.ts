import styled, {keyframes} from 'styled-components';

interface IApp {
  'data-escolhido': boolean;
}

const spin = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

export const App = styled.div`
  text-align: center;
`;

export const Botao = styled.button<IApp>`
  width: 100px;
  height: 30px;
  margin: 10px;
  background-color: ${props => (props['data-escolhido'] ? '#aff' : '#aaa')};
`;

export const ContainerBotoes = styled.div`
  margin-top: 20px;
  margin-bottom: 50px;
`;

export const Header = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  height: 100px;
`;

export const Item = styled.a`
  margin: 10px;
`;

export const Link = styled.a`
  color: #61dafb;
`;

export const Logo = styled.img`
  height: 40vmin;
  pointer-events: none;

  @media (prefers-reduced-motion: no-preference) {
    animation: ${spin} infinite 20s linear;
  }
`;

export const Main = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
`;

export const Nav = styled.nav`
  display: flex;
  padding-right: 30px;
`;

export const Texto = styled.p``;

export const TextoHeader = styled.p`
  font-size: 15px;
  color: #fff;
  cursor: pointer;
`;
