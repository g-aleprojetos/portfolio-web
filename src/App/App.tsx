import React from 'react';
import logo from '../assets/images/logo.svg';
import {useTranslation} from 'react-i18next';
import {namespaces} from 'utils/i18n/i18n.constants';
import '../utils/i18n';
import * as S from './App.styles';

export const App = () => {
  const {t, i18n} = useTranslation(namespaces.pages.header);
  return (
    <S.App>
      <S.Header>
        <S.Nav>
          <S.Item>
            <S.TextoHeader>{t('home')}</S.TextoHeader>
          </S.Item>
          <S.Item>
            <S.TextoHeader>{t('contents')}</S.TextoHeader>
          </S.Item>
          <S.Item>
            <S.TextoHeader>{t('about')}</S.TextoHeader>
          </S.Item>
        </S.Nav>
      </S.Header>
      <S.Main>
        <S.ContainerBotoes>
          <S.Botao
            data-escolhido={i18n.resolvedLanguage === 'pt'}
            type="submit"
            onClick={() => i18n.changeLanguage('pt')}>
            PortuguÃªs
          </S.Botao>
          <S.Botao
            data-escolhido={i18n.resolvedLanguage === 'en'}
            type="submit"
            onClick={() => i18n.changeLanguage('en')}>
            English
          </S.Botao>
        </S.ContainerBotoes>
        <S.Logo src={logo} alt="logo" />
        <S.Texto>
          Edit <code>src/App.tsx</code> and save to reload.
        </S.Texto>
        <S.Link
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer">
          Learn React
        </S.Link>
      </S.Main>
    </S.App>
  );
};
