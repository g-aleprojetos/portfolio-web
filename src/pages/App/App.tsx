import React from 'react';
import logo from 'assets/images/logo.svg';
import {useTranslation} from 'react-i18next';
import {namespaces} from 'utils/i18n/i18n.constants';
import 'utils/i18n';
import {ToggleSwitch} from 'components/toggleSwitch';
import * as S from './App.styles';
import {useBackgroundContext} from 'context/background';

export const App = () => {
  const {themeDark, handleToggle} = useBackgroundContext();
  const {t, i18n} = useTranslation(namespaces.pages.header);

  return (
    <S.App>
      <S.Header>
        <S.Nav>
          <S.Item>
            <S.TextoHeader>{t('home')}</S.TextoHeader>
          </S.Item>
          <S.Item>
            <S.TextoHeader>{t('contents')}</S.TextoHeader>
          </S.Item>
          <S.Item>
            <S.TextoHeader>{t('about')}</S.TextoHeader>
          </S.Item>
          <ToggleSwitch isOn={themeDark} handleOnclick={handleToggle} />
        </S.Nav>
      </S.Header>
      <S.Main>
        <S.ContainerBotoes>
          <S.Botao
            data-escolhido={i18n.resolvedLanguage === 'pt'}
            type="submit"
            onClick={() => i18n.changeLanguage('pt')}>
            Português
          </S.Botao>
          <S.Botao
            data-escolhido={i18n.resolvedLanguage === 'en'}
            type="submit"
            onClick={() => i18n.changeLanguage('en')}>
            English
          </S.Botao>
          <S.Botao
            data-escolhido={i18n.resolvedLanguage === 'fr'}
            type="submit"
            onClick={() => i18n.changeLanguage('fr')}>
            Français
          </S.Botao>
        </S.ContainerBotoes>
        <S.Logo src={logo} alt="logo" />
        <S.Texto>
          Edit <code>src/App.tsx</code> and save to reload.
        </S.Texto>
        <S.Link
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer">
          Learn React
        </S.Link>
      </S.Main>
    </S.App>
  );
};
