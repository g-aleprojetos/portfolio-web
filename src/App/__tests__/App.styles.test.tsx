import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as S from '../App.styles';

describe('App.styles', () => {
  test('App DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.App />);
    expect(container).toMatchSnapshot();
  });

  test('Header DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Header />);
    expect(container).toMatchSnapshot();
  });

  test('Link DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Link />);
    expect(container).toMatchSnapshot();
  });

  test('Logo DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Logo />);
    expect(container).toMatchSnapshot();
  });

  test('Texto DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Texto />);
    expect(container).toMatchSnapshot();
  });
});
