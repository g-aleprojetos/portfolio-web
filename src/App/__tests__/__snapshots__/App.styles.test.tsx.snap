// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.styles App DEVE ser igual ao snapshot 1`] = `
.c0 {
  text-align: center;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`App.styles Header DEVE ser igual ao snapshot 1`] = `
.c0 {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

<div>
  <header
    class="c0"
  />
</div>
`;

exports[`App.styles Link DEVE ser igual ao snapshot 1`] = `
.c0 {
  color: #61dafb;
}

<div>
  <a
    class="c0"
  />
</div>
`;

exports[`App.styles Logo DEVE ser igual ao snapshot 1`] = `
.c0 {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .c0 {
    animation: fvtopB infinite 20s linear;
  }
}

<div>
  <img
    class="c0"
  />
</div>
`;

exports[`App.styles Texto DEVE ser igual ao snapshot 1`] = `
<div>
  <p
    class=""
  />
</div>
`;
